tests/main.test.ts

```diff
@@ -47,22 +47,66 @@ describe("src/main.ts", function() {
 
     const validAny = validModel(anyType);
 
-    it('should allow all types when no restriction', function(done) {
+    it('should allow all types for -any- schema when there are no restriction',
+    function(done) {
       const validTypes = [
         undefined,
         null,
         5,
         "string",
         true,
-        {first: 1, second: "two"},
-        // null,
-        // undefined
+        {first: 1, second: "two"}
       ];
 
-      mochaAssertPromiseResovles(
-        Promise.all(validTypes.map(validAny)),
+      mochaAssertPromiseResovles(Promise.all(validTypes.map(validAny)), done);
+    });
+
+    const anyWithBoolRestriction: anySchema = {
```

Comment:
Doesn't have to be a `bool`, better to name this `anyWithTruthyRestriction`. Lots of things other than bools will pass this schema. Same for `someBool` name used below.